File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-7.java
class Solution {
    public int search(int[] nums, int target) {
        
        int beg = 0, end = nums.length - 1;
        int ans = -1;

        while(beg <= end)
        {
            int mid = beg + (end - beg)/2;
            if(nums[mid] == target)
            {
                ans = mid;
                break;
            }
            if(nums[beg] < nums[end])
            {
                
                if(nums[mid] > target)
                    end = mid - 1;
                else
                    beg = mid + 1;
            }
            else
            {
                if(nums[mid] >= nums[beg])
                {
                    if(target >= nums[beg] && target < nums[mid])
                        end = mid - 1;
                    else
                        beg = mid + 1;
                }
                else
                {
                    if(target > nums[mid] && target <= nums[end])
                        beg = mid + 1;
                    else
                        end = mid - 1;
                }
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-7.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        zeroCount,n = 0,len(nums)
        l,r = 0,0
        ans = 0

        while r < n:
            if nums[r] == 0:
                zeroCount += 1
            #Check if zeroCount is > k then slide the widnow
            while l <= r and zeroCount > k:
                if nums[l] == 0:
                    zeroCount -= 1
                l += 1
            #Take the max as ans
            ans = max(ans,r-l+1)
            r += 1
        
        return ans
                
            